src/app/services/batch-service/batch.service.spec.ts

```diff
@@ -15,60 +15,93 @@ import { MyCarComponent } from 'src/app/components/my-car/my-car.component';
+import { BsNavbarComponent } from 'src/app/bs-navbar/bs-navbar.component';
+import { ScheduleComponent } from 'src/app/components/schedule/schedule.component';
+import { ViewMyRidesComponent } from 'src/app/components/view-my-rides/view-my-rides.component';
 
 
 describe('BatchService', () => {
   let batchService: BatchService;
   
   beforeEach(() => {
     TestBed.configureTestingModule({
-    declarations: [AdminComponent, CarRegisterComponent, UserRegisterComponent, LoginComponent, MyCarComponent, NavbarComponent, PreferenceComponent, ProfileComponent],
-    imports: [HttpClientModule, AppRoutingModule, FormsModule],
+      declarations: [
+        RegisterComponent,
+        DriverComponent,
+        AdminComponent,
+        LoginComponent,
+        CarRegisterComponent,
+        LoginComponent,
+        NavbarComponent,
+        MyCarComponent,
+        ProfileComponent,
+        PreferenceComponent,
+        AdminLoginComponent,
+        DriverInfoComponent,
+        LandingComponent,
+        SignupModalComponent,
+        LandingPageComponent,
+        ProfileContactComponent,
+        ProfileCarComponent,
+        ProfileLocationComponent,
+        ProfileMembershipComponent,
+        DriverContactModalComponent,
+        DriverListComponent,
+        UserRegisterComponent,
+        HomePageComponent,
+        BsNavbarComponent,
+        ScheduleComponent,
+        ViewMyRidesComponent,
+      ],
+      imports: [HttpClientModule, AppRoutingModule, FormsModule, ReactiveFormsModule],
     providers: [{provide: APP_BASE_HREF, useValue: '/my/app'}]
   })
 
   batchService = TestBed.get(BatchService);
 })
 
+it('should be created', () => {
+  const service: BatchService = TestBed.get(BatchService);
+  expect(service).toBeTruthy();
+});
+
 it('should register a batch', () => {
   expect(batchService).toBeTruthy();
 });
 
-//  //Adding test for getAllBatches() method
-//  describe('getAllBatches', () => {
-//   it('should return a list of batches', () => {
-//     const batchResponse = [
-//       {
-//         batchNumber: 1,
-//         batchLocation: 'NYC'
-//       },
-//       {
-//         batchNumber: 2,
-//         batchLocation: 'VA'
-//       }
-//     ];
-//     let response;
-//     spyOn(batchService, 'getAllBatches').and.returnValue(of(batchResponse));
+ //Adding test for getAllBatches() method
+ describe('getAllBatches', () => {
+  it('should return a list of batches', () => {
+    const batchResponse: Batch[] = [
+      {
+        batchNumber: 1,
+        batchLocation: 'NYC'
+      },
+      {
+        batchNumber: 2,
+        batchLocation: 'VA'
+      }
+    ];
+    let response: Batch[];
+    spyOn(batchService, 'getAllBatches').and.returnValue(batchResponse);
 
-//     // batchService.getAllBatches().subscribe(res => {
-//     //   response = res;
-//     // });
+    response = batchService.getAllBatches();
 
-//     expect(response).toEqual(batchResponse);
-//   });
-// });
+    expect(response).toEqual(batchResponse);
```

Comment:
Does this unit test actually do anything?

Reply:
That test verifies that the get all batches method is returning, but since it uses a spy on to control the response, it's a bit of a smoke test. I don't know why it was commented out, but it passes so I don't think it should be removed, but I admit it is a little silly.

Reply:
Well, it is basically asserting that custom object is equal to itself.

Reply:
It passes because we basically just have an empty unit test (barring an exception being thrown).

Reply:
Okay, I'll take it out, no problem.